src/main/java/com/volasoftware/tinder/service/UserService.java

```diff
@@ -135,4 +135,19 @@ public UserProfileDto getCurrentUserProfile() {
 
         return new UserProfileDto(user.getFirstName(), user.getLastName(), user.getEmail(), user.getGender());
     }
+
+    public UserProfileDto editUserProfile(@RequestBody UserProfileDto userProfileDto){
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        String currentUser = authentication.getName();
+
+        User user = userRepository.findOneByEmail(currentUser).orElseThrow(() ->
+                new NotLoggedInException("You are not logged in!"));
+        user.setFirstName(userProfileDto.getFirstName());
+        user.setLastName(userProfileDto.getLastName());
+        user.setEmail(userProfileDto.getEmail());
+        user.setGender(userProfileDto.getGender());
```

Comment:
Are all fields required in UserProfileDto, because these setters will override the values? Be careful what we want and possible values that can be provided in the request.